dev/styler_utils.R

```diff
@@ -178,9 +178,90 @@ style_files = function(
 
 #' rpolars style entire package
 #' @return NULL
+#' @examples
 #' style_entire_pkg()
 style_entire_pkg = function() {
   list.files(".", full.names = TRUE, pattern = "\\.R$|\\.Rmd$", recursive = TRUE) |>
     setdiff("./R/extendr-wrappers.R") |>
     style_files()
 }
+
+#' Update the text in the usage section of an Rd file
+#'
+#' @param file path of an Rd file
+#' @param pattern Pattern to look for
+#' @param replacement Regex value
+#' @return The updated Rd file content invisibly
+#'
+#' @examples
+#' rd_usage_replace("man/DataFrame_clone.Rd", "DataFrame_", "<DataFrame>$")
+rd_usage_replace = function(file, pattern, replacement) {
+  content_rd = tools::parse_Rd(file)
+  tags = tools:::RdTags(content_rd)
+  content_rd[[which(tags == r"(\usage)")]][[2]][1] = content_rd[[which(tags == r"(\usage)")]][[2]][1] |>
+    stringr::str_replace_all(pattern, replacement)
+
+  content_rd |>
+    as.character(deparse = TRUE) |>
+    paste0(collapse = "") |>
+    brio::write_file(file)
+}
+
+#' Update all usage sections in Rd files
+#' @param files A character vector of Rd files to update. If NULL, all Rd files in the package are updated.
+#' @return NULL
+#' @examples
+#' rd_usage_update()
+rd_usage_update = function(files = NULL) {
+  files = files %||% list.files("man", pattern = r"(.*\.Rd$)", full.names = TRUE, recursive = TRUE)
+
+  patterns = tibble::tribble(
+    ~pattern, ~replacement,
+    "DataFrame_", "<DataFrame>$",
+    "DynamicGroupBy_", "<DynamicGroupBy>$",
+    "Expr_", "<Expr>$",
+    "ExprBin_", "<Expr>$bin$",
+    "ExprCat_", "<Expr>$cat$",
+    "ExprDT_", "<Expr>$dt$",
+    "ExprList_", "<Expr>$list$",
+    "ExprMeta_", "<Expr>$meta$",
+    "ExprName_", "<Expr>$name$",
+    "ExprStr_", "<Expr>$str$",
+    "ExprStruct_", "<Expr>$struct$",
+    "GroupBy_", "<GroupBy>$",
+    "(IO|pl)_read_", "pl$read_", # file names are "IO_read_", function names are "pl_read_"
+    "(IO|pl)_scan_", "pl$scan_", # file names are "IO_scan_", function names are "pl_scan_"
+    "(IO|LazyFrame)_sink_", "<LazyFrame>$sink_", # file names are "IO_sink_", function names are "LazyFrame_sink_"
+    "(IO|DataFrame)_write_", "<DataFrame>$write_", # file names are "IO_write_", function names are "DataFrame_write_"
+    "LazyFrame_", "<LazyFrame>$",
+    "LazyGroupBy_", "<LazyGroupBy>$",
+    "pl_", "pl$",
+    "RField_", "<RField>$",
+    "RThreadHandle_", "<RThreadHandle>$",
+    "Series_", "<Series>$",
+    "SQLContext_", "<SQLContext>$",
+  )
+
+  purrr::walk2(
+    patterns$pattern, patterns$replacement,
+    \(pattern, replacement) {
+      files |>
+        stringr::str_subset(pattern) |>
+        purrr::walk(
+          \(file) {
+            tryCatch(
+              rd_usage_replace(file, pattern, replacement),
+              error = function(e) {
+                cat("Error updating usage section in ", file, "\n", sep = "")
+              }
+            )
+          }
+        )
+    }
+  )
+}
```

Comment:
@etiennebacher I think these functions will handle updates to the Usage section better than the current script for altdoc, but it requires storing the Rd file in a separate directory.
If you can change the directory altdoc refers to, you can switch to using these functions.

Reply:
What is better handled by this script?

Reply:
Like this:
https://github.com/pola-rs/r-polars/blob/1f2a4d0494a0e90e49ccb72cc92e10bd82a0a42d/man/pl_options.Rd#L7-L19

![image](https://github.com/pola-rs/r-polars/assets/50911393/1bbc740e-a1a9-4cff-856d-b3f6092e6c2d)

Reply:
Ah yes, indeed, I'll improve the postprocessing script with yours